server/model/promo.js

```diff
@@ -0,0 +1,11 @@
+import {Record} from 'immutable';
+
+export const Promo = Record({
+  modifiers: null,
+  url: null,
+  image: null,
+  meta: null,
+  title: null,
+  copy: null,
+  type: null
+});
```

Comment:
I've pushed as close to what we were thinking of as possible – not sure it's close/good enough – will see about updating with the type-checking stuff @jamesgorrie has been working on as and when that gets in and mark this as WIP for now.